modules/server/src/network/index.ts

```diff
@@ -1,10 +1,111 @@
+	!!input && typeof input === 'object' && '__schema' in input && typeof input.__schema === 'object';
+
+type FetchRemoteSchemaResult = {
+	config: NetworkAggregationConfigInput;
+	introspectionResult: IntrospectionQuery | null;
+};
+
+/**
+ *
+ * @param config - network config from env
+ * @returns a promise containing network config and the introspection query result
+ *
+ * @throws Fetch failed error
+ *
+ * @throws JSON parse error
+ *
+ * @throws Unexpected data error eg. Not a valid introspectionQuery result
+ */
+const fetchRemoteSchema = async (
+	config: NetworkAggregationConfigInput,
+): Promise<FetchRemoteSchemaResult> => {
+	const { graphqlUrl } = config;
+	try {
+		/**
+		 * get full schema (needed for buildClientSchema) and convert json
+		 */
+		const response = await fetchGql({
+			url: graphqlUrl,
+			gqlRequest: { query: getIntrospectionQuery() },
+		});
+
+		if (response.status === 200 && response.statusText === 'OK') {
+			// axios response "data" field, graphql response "data" field
+			const responseData = response.data?.data;
+
+			if (isGqlIntrospectionQuery(responseData)) {
+				return { config, introspectionResult: responseData };
+			} else {
+				console.error('unexpected response data in fetchRemoteSchema');
+				throw Error('unexpected response data');
+			}
+		} else {
+			console.error('network error in fetchRemoteSchema');
+			throw Error('network error');
+		}
+	} catch (error) {
+		console.log(`failed to retrieve schema from url: ${config.graphqlUrl}`);
+		console.error(error);
+		return { config, introspectionResult: null };
+	}
+};
+
+const fetchRemoteSchemas = async ({
 	networkConfig,
 }: {
-	networkConfig: NetworkAggregationInterface[];
+	networkConfig: NetworkAggregationConfigInput[];
+}): Promise<NetworkAggregationConfig[]> => {
+	// query remote notes
+	const networkQueryPromises = networkConfig.map(async (config) => fetchRemoteSchema(config));
+	// type cast because rejected errors are handled
+	const networkQueries = (await Promise.allSettled(
+		networkQueryPromises,
+	)) as PromiseFulfilledResult<FetchRemoteSchemaResult>[];
+
+	// build schema
+	const schemaResults = networkQueries.map((networkResult) => {
+		const { config, introspectionResult } = networkResult.value;
+
+		try {
+			const schema = introspectionResult !== null ? buildClientSchema(introspectionResult) : null;
+			return { ...config, schema };
+		} catch (error) {
+			console.error('build schema error', error);
+			return { ...config, schema: null };
+		}
+	});
+
+	return schemaResults;
+};
+
+export const createSchemaFromNetworkConfig = async ({
+	networkConfig,
+}: {
+	networkConfig: NetworkAggregationConfigInput[];
 }) => {
-	console.log('network config', networkConfig);
+	const remoteSchemasResult = await fetchRemoteSchemas({
+		networkConfig,
+	});
+
+	const mergedFields = mergeRemoteSchemas(remoteSchemasResult);
+};
+
+const mergeRemoteSchemas = (input) => {
+	/**placeholder */
```

Comment:
This placeholder is telling other devs reading this nothing about this function, I would want more information here before being able to review if this is the correct step to use. As a placeholder, lets at least explicitly declare `(input: any): any => {` so TS will not complain about implicit any.

It will benefit this entire process to be strict with our types. Define the expected inputs. Declare the expected output. Start with the type signatures and everything else will just be the logic to convert inputs to outputs. If the specific types are still unclear, than use the `any` suggestion above. In either case, give me a function comment or description to explain the placeholder.